### ISSUE TITLE
Provide verbose logging mode for Argo CLI

### ISSUE THREAD

**Issue by thundergolfer** [argoproj/argo#1666]
**Is this a BUG REPORT or FEATURE REQUEST?**:

FEATURE REQUEST

-----

**TLDR:** Like `kubectl`'s `-v` option, `argo` should provide verbose logging optionally so users can inspect the API requests being made.

We're running Argo workflows on EKS and we're currently experiencing significant performance issues. `argo list` operations often take > 1min to return or just timeout altogether. This is in a namespace with ~ 100 workflows total.

In release `2.4.0` the `--chunk-size` option from `kubectl` was put in `argo`, and that certainly has helped reduce timeouts but `argo list` operations are still _extremely_ slow.

It'd be great to be able to see detail of what's happening in the time a `list` operation is being executed. 

-----

**Environment**:
- Argo version: `2.4.1`

**Comment by trondhindenes** [comment]
imho this is almost a must in order to be able to interact with workflows from other apps - we need to be able to see the api calls made in order to be able to emulate them.

**Comment by alexec** [comment]
Inspecting requests can be done using Wireshark or similar. That said - in v2.4 - all API requests are basically calls to Kubernetes, e.g. `argo list` is same as `kubectl get workflows`. Might be enough to get your data.

**Comment by thundergolfer** [comment]
Thanks for the reply.

>  e.g. argo list is same as kubectl get workflows.

Is this true for all `argo` CLI commands, and will it be true going forward? If the Argo CLI is making different requests to `kubectl` in any way, it will be useful to use `--verbose` to see exactly how they differ in the event of performance differences.

`--verbose` likely has other uses too.

(Can say though our performance problems with Argo have passed 🙂)

**Comment by alexec** [comment]
This should be fixed now, try:

```
argo --loglevel DEBUG submit ...
```

**Comment by alexec** [comment]
This produces debugging if you're running Argo Server:
```
GODEBUG=http2debug=2 argo submit examples/hello-world.yaml
```

**Comment by alexec** [comment]
I'm closing this as the solution meets the description. However, printing the requests does not happen (it is GRPC anyway - not HTTP+REST). Feel free to open a new issue.
